src/dev/LCD.cpp

```diff
@@ -0,0 +1,95 @@
+    this->regSelect.writePin(EVT::core::IO::GPIO::State::LOW);
+    this->reset.writePin(EVT::core::IO::GPIO::State::LOW);
+    this->spi = spi;
+}
+
+void LCD::dataWrite(unsigned char data) {
+    this->regSelect.writePin(EVT::core::IO::GPIO::State::HIGH);
+    this->spi.startTransmission(0);
+    this->spi.write(&data, 1);
+    this->spi.endTransmission(0);
+}
+
+void LCD::commWrite(unsigned char data) {
+    this->regSelect.writePin(EVT::core::IO::GPIO::State::LOW);
+    this->spi.startTransmission(0);
+    this->spi.write(&data, 1);
+    this->spi.endTransmission(0);
+}
+
+void LCD::drivePixel(unsigned char page, unsigned char col_up, unsigned char col_low, unsigned char data) {
+    this->commWrite(0x40);          //line to start writing on (0 -> 64) moves set bits with it DO NOT CHANGE
+    this->commWrite(0xB0 + page);   //writes the page address (4 bits, 8 rows selcted by values 0-7 )
+    this->commWrite(0x10 + col_up); //writes the first 4 bits of the column select (out of 8 bits)
+    this->commWrite(0x00 + col_low);//writes the second 4 bits of the column select (out)
+
+    this->dataWrite(data);//writes 8 vertical bits based on value between 0-255 based on bits set ex: 01001100(0x4C) is |WHITE|
+                          //                                                                                                 |BLACK|
+                          //                                                                                                 |WHITE|
+                          //                                                                                                 |WHITE|
+                          //                                                                                                 |BLACK|
+                          //                                                                                                 |BLACK|
+                          //                                                                                                 |WHITE|
+                          //                                                                                                 |WHITE|
+}
+
+void LCD::clearLCD(unsigned char* bitMap) {
+    unsigned int i, j;
+    unsigned char page = 0xB0;
+    this->commWrite(0xAE);         //Display OFF
+    this->commWrite(0x40);         //Display start address + 0x40
+    for (i = 0; i < 8; i++) {      //64 pixel display / 8 pixels per page = 8 pages
+        this->commWrite(page);     //send page address
+        this->commWrite(0x10);     //column address upper 4 bits + 0x10
+        this->commWrite(0x00);     //column address lower 4 bits + 0x00
+        for (j = 0; j < 128; j++) {//128 columns wide
+            this->dataWrite(0x00); //write clear pixels
+            bitMap++;
+        }
+        page++;//after 128 columns, go to next page
+    }
+    this->commWrite(0xAF);
+}
+
+void LCD::displayMap(unsigned char* bitMap) {
```

Comment:
This also feels like there should be a sequential write option instead of just specifying one pixel after the next. Leave it for now, but we should investigate this